lib/edmunds/vehicle/specification/style.rb

```diff
@@ -9,13 +9,21 @@ module Vehicle
     module Specification
       module Style
         class Style
-          attr_reader :id, :name, :trim, :body, :year, :make_name, :model_name
+          attr_reader :id, :name, :trim, :body, :year, :make_name, :model_name, :engine, :transmission, :driven_wheels, :colors, :options
 
-          def initialize(attributes)
+          def initialize(attributes, view = nil)
```

Comment:
https://github.com/Sovietaced/edmunds/blob/master/lib/edmunds/vehicle/specification/make.rb#L19-L36

I have completed the refactor that supported optional params. In this case "view" is an optional parameter. 

Instead of instantiating the request and storing the "view" enum I would just handle engine, transmission, driven_wheels if there are found in the hash.